pkg/controller/nodesetup/sync_raids.go

```diff
@@ -130,3 +138,63 @@ func filterMatchingRe(blockDevices []*blkutils.BlockDevice, nameRegexp, modelReg
 
 	return devices, nil
 }
+
+type blockDevice struct {
+	blkutils.BlockDevice
+	APIName string
+}
+
+// listBlockDevices returns a list of block devices on host together with names under which they were referenced on API object.
+func listBlockDevices(ctx context.Context, executor exec.Interface, nc *scyllav1alpha1.NodeConfig, devfsPath string) ([]*blockDevice, []metav1.Condition, error) {
+	bds, err := blkutils.ListBlockDevices(ctx, executor)
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't list block devices: %w", err)
+	}
+
+	var progressingConditions []metav1.Condition
+	blockDevices := make([]*blockDevice, 0, len(bds))
+
+	for _, bd := range bds {
+		var apiName string
+		for _, lc := range nc.Spec.LocalDiskSetup.LoopDevices {
+			symlinkPath := getLoopDeviceSymlinkPath(devfsPath, lc.Name)
+
+			_, err := os.Stat(symlinkPath)
+			if err != nil {
+				if errors.Is(err, fs.ErrNotExist) {
+					progressingConditions = append(progressingConditions, metav1.Condition{
+						Type:               raidControllerNodeProgressingConditionFormat,
+						Status:             metav1.ConditionTrue,
+						Reason:             "AwaitingLoopDevice",
+						Message:            fmt.Sprintf("Loop device %q not created yet", lc.Name),
+						ObservedGeneration: nc.Generation,
+					})
+					continue
+				}
+
+				return nil, progressingConditions, fmt.Errorf("can't stat path %q: %w", symlinkPath, err)
```

Comment:
both are correct, but just to follow up on our conversation in https://github.com/scylladb/scylla-operator/pull/1691#discussion_r1476439609 this might look better if the shorter branch gets terminated first (I don't mind either, so no action is needed here, unless that's what you prefer)
```suggestion
				if !errors.Is(err, fs.ErrNotExist) {
					return nil, progressingConditions, fmt.Errorf("can't stat path %q: %w", symlinkPath, err)			
				}

				progressingConditions = append(progressingConditions, metav1.Condition{
					Type:               raidControllerNodeProgressingConditionFormat,
					Status:             metav1.ConditionTrue,
					Reason:             "AwaitingLoopDevice",
					Message:            fmt.Sprintf("Loop device %q not created yet", lc.Name),
					ObservedGeneration: nc.Generation,
				})
				continue
```